### PR TITLE
[pipes] use PipesDefaultLogWriter for AWS EMR

### PR DESCRIPTION
**Pull request by danielgafni**
## Summary & Motivation

Add a new `include_stdio_in_messages` parameter to `PipesS3MessageReader`. It enables log forwarding to Dagster via Pipes messages. 

## How I Tested These Changes

Executed a real AWS EMR job and confirmed we are getting the logs

## Changelog

[dagster-pipes, dagster-aws] `PipesS3MessageReader` now has a new parameter `include_stdio_in_messages` which enables log forwarding to Dagster via Pipes messages.

### CODE REVIEW

**Comment by danielgafni** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25826?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25826** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25826?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#25679** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25679?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#25770](https://github.com/dagster-io/dagster/pull/25770) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25770?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#25764** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25764?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25564** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25564?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#25574](https://github.com/dagster-io/dagster/pull/25574) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25574?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#25572** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25572?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @danielgafni and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/emr.py [hunk 1]
```diff
@@ -358,7 +367,9 @@ def _extract_dagster_metadata(
             metadata["AWS EMR Cluster ID"] = cluster_id
 
         if log_uri := cluster.get("LogUri"):
-            metadata["AWS EMR Log URI"] = MetadataValue.path(log_uri)
+            # LogUri originally points to a shared S3 bucket where logs from all clusters are stored
+            # which is not very useful
+            metadata["AWS EMR Log URI"] = MetadataValue.path(log_uri + f"{cluster_id}")
```
Comment (danielgafni) [inline python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/emr.py:372]: drive-by improvement

**Review by smackesey — APPROVED** [review]
